### ISSUE TITLE
Slow query 200k+

### ISSUE THREAD

**Issue by xxx2134** [woocommerce/woocommerce#12524]
```
SELECT SQL_CALC_FOUND_ROWS wp_posts.ID
FROM wp_posts
INNER JOIN wp_postmeta
ON ( wp_posts.ID = wp_postmeta.post_id )
WHERE 1=1
AND ( ( wp_postmeta.meta_key = '_visibility'
AND wp_postmeta.meta_value IN ('visible','catalog') ) )
AND wp_posts.post_type = 'product'
AND (wp_posts.post_status = 'publish'
OR wp_posts.post_status = 'private')
GROUP BY wp_posts.ID
ORDER BY wp_posts.menu_order ASC, wp_posts.post_title ASC
LIMIT 0, 10;
```

Time execute 10 sec+!!!!!!!!!!!!

HELP ME!

**Comment by mikejolley** [comment]
Not much detail here? Shared host? Product count? System status?

**Comment by xxx2134** [comment]
wp_posts ~185,014 records
wp_postmeta ~ 685,083

```
SELECT SQL_CALC_FOUND_ROWS wp_posts.ID
FROM wp_posts
INNER JOIN wp_postmeta
ON ( wp_posts.ID = wp_postmeta.post_id )
WHERE 1=1
AND ( ( wp_postmeta.meta_key = '_visibility'
AND wp_postmeta.meta_value IN ('visible','catalog') ) )
AND wp_posts.post_type = 'product'
AND (wp_posts.post_status = 'publish'
OR wp_posts.post_status = 'private')
GROUP BY wp_posts.ID

LIMIT 0, 10;
```

10.7711 sec

```
SELECT wp_posts.ID
FROM wp_posts
INNER JOIN wp_postmeta ON ( wp_posts.ID = wp_postmeta.post_id )
WHERE 1 =1
AND (
(
wp_postmeta.meta_key = '_visibility'
AND wp_postmeta.meta_value
IN (
'visible', 'catalog'
)
)
)
AND wp_posts.post_type = 'product'
AND (
wp_posts.post_status = 'publish'
OR wp_posts.post_status = 'private'
)
GROUP BY wp_posts.ID
LIMIT 0 , 10
```
0.0214 sec

**Comment by johnrom** [comment]
(I could be wrong but...) It appears to me as though WC is doing the exact query it is supposed to do, and for that many records WP Query alone isn't cutting it. When you have a site of that size there will often be customization needed to maintain adequate performance. What you've done to the query to cut down the time would not be suitable for everyone using WooCommerce, nor would abandoning WP_Query. Overall this appears to be a support question and not a code issue.

You may be able to circumvent this entire issue by tuning your database, possibly adding an index on `menu_order` and `menu_title` (include both in the index, in that order). Otherwise, you can choose to remove `SQL_CALC_FOUND_ROWS` -- which would remove pagination's ability to know what the last page is -- and/or the sorting yourself (e.,g., sort only alphabetically, or sort only by menu order). These are possible by filtering `pre_get_posts` to set `no_found_rows` to true and setting the `orderby` to whichever you'd like on `WP_Query`, or even by post ID for the fastest possible time.

The only way this would be a code issue is to debate whether there should be a threshold whereby this is automatically implemented. Say, 100,000 products = remove SQL_CALC_FOUND_ROWS if a user checks a setting like "Tune Performance for Large Inventories (with a loss of functionality)", etc

**Comment by mikejolley** [comment]
I've found some ways to significantly speed up queries using terms instead of meta. I'm submitting a proof of concept PR soon.

**Comment by mikejolley** [comment]
Closed in favour of #12527
